catalog/resource_grant_test.go

```diff
@@ -0,0 +1,732 @@
+		RequiresNew: false, // test the negative as the positive is not covered by qa.ResourceFixture
+		ID:          "table/foo.bar.baz/me",
+		InstanceState: map[string]string{
+			"table":     "foo.bar.baz",
+			"principal": "someone-else",
+		},
+		HCL: `
+		table = "foo.bar.baz"
+
+		principal = "me"
+		privileges = ["MODIFY", "SELECT"]
+		`,
+	}.ExpectError(t, "changes require new: principal")
+}
+
+func TestResourceGrantDelete(t *testing.T) {
+	qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "GET",
+				Resource: "/api/2.1/unity-catalog/permissions/table/foo.bar.baz?",
+				Response: catalog.PermissionsList{
+					PrivilegeAssignments: []catalog.PrivilegeAssignment{
+						{
+							Principal:  "me",
+							Privileges: []catalog.Privilege{"MODIFY", "SELECT"},
+						},
+						{
+							Principal:  "someone-else",
+							Privileges: []catalog.Privilege{"MODIFY", "SELECT"},
+						},
+					},
+				},
+			},
+			{
+				Method:   "PATCH",
+				Resource: "/api/2.1/unity-catalog/permissions/table/foo.bar.baz",
+				ExpectedRequest: catalog.UpdatePermissions{
+					Changes: []catalog.PermissionsChange{
+						{
+							Principal: "me",
+							Remove:    []catalog.Privilege{"MODIFY", "SELECT"},
+						},
+					},
+				},
+			},
+			{
+				Method:   "GET",
+				Resource: "/api/2.1/unity-catalog/permissions/table/foo.bar.baz?",
+				Response: catalog.PermissionsList{
+					PrivilegeAssignments: []catalog.PrivilegeAssignment{
+						{
+							Principal:  "someone-else",
+							Privileges: []catalog.Privilege{"MODIFY", "SELECT"},
+						},
+					},
+				},
+			},
+		},
+		Resource: ResourceGrant(),
+		Delete:   true,
+		ID:       "table/foo.bar.baz/me",
+		InstanceState: map[string]string{
+			"table":     "foo.bar.baz",
+			"principal": "me",
+		},
+		HCL: `
+		table = "foo.bar.baz"
+
+		principal = "me"
+		privileges = ["MODIFY", "SELECT"]
+		`,
+	}.ApplyNoError(t)
+}
+
+func TestResourceGrantReadMalformedId(t *testing.T) {
+	qa.ResourceFixture{
+		Resource: ResourceGrant(),
+		ID:       "foo.bar",
+		Read:     true,
+		HCL: `
+		table = "foo"
+		principal = "me"
+		privileges = ["MODIFY", "SELECT"]
+		`,
+	}.ExpectError(t, "ID must be three elements split by `/`: foo.bar")
+}
+
+func TestResourceGrantCreateNoSecurable(t *testing.T) {
+	qa.ResourceFixture{
+		Resource: ResourceGrant(),
+		Create:   true,
+		HCL: `
+		principal = "me"
+		privileges = ["MODIFY", "SELECT"]
+		`,
+	}.ExpectError(t, "invalid config supplied. [catalog] Missing required argument. [external_location] Missing required argument. [foreign_connection] Missing required argument. [function] Missing required argument. [metastore] Missing required argument. [model] Missing required argument. [pipeline] Missing required argument. [recipient] Missing required argument. [schema] Missing required argument. [share] Missing required argument. [storage_credential] Missing required argument. [table] Missing required argument. [volume] Missing required argument")
```

Comment:
Oof... We don't have any control over this error message do we?

Reply:
Unfortunately not. I did the best I could by using `ConflictsWith` to trim the error down for the `TestResourceGrantCreateOneSecurableOnly` case, but can't do anything here.